File: examples/js/risk-score-job/job.js
import { SchedulerJob, runJob } from "./scheduler-sdk.js";

export default class RiskScoreJob extends SchedulerJob {
  async execute(context) {
    const scanType    = context.param.requiredString("scanType");
    const applications = context.param.requiredStringArray("applications");

    const response = await context.igaClient.execute(
      "POST",
      "/scheduler/risk-scores/recompute",
      { scanType, applications }
    );

    return {
      status: "submitted",
      igaRequestId: response.requestId
    };
  }
}

// Only auto-run when executed directly (not when imported by tests)
if (!process.env.VITEST) runJob(RiskScoreJob);

File: examples/python/risk-score-job/job.py
from iga_scheduler import SchedulerJob, run_job


class RiskScoreJob(SchedulerJob):
    def execute(self, context):
        scan_type = context["param"].required_string("scanType")
        applications = context["param"].required_string_array("applications")
        response = context["iga_client"].execute(
            "POST",
            "/scheduler/risk-scores/recompute",
            {"scanType": scan_type, "applications": applications},
        )
        return {"status": "submitted", "igaRequestId": response.get("requestId")}


run_job(RiskScoreJob)
